edgelet/iotedge/src/check/checks/container_local_time.rs

```diff
@@ -0,0 +1,71 @@
+use failure::{self, Context, ResultExt};
+use std::time::Duration;
+
+use crate::check::{checker::Checker, Check, CheckResult};
+
+#[derive(Default, serde_derive::Serialize)]
+pub(crate) struct ContainerLocalTime {
+    expected_duration: Option<Duration>,
+    actual_duration: Option<Duration>,
+    diff: Option<u64>,
+}
+
+impl Checker for ContainerLocalTime {
+    fn id(&self) -> &'static str {
+        "container-local-time"
+    }
+    fn description(&self) -> &'static str {
+        "container time is close to host time"
+    }
+    fn execute(&mut self, check: &mut Check) -> CheckResult {
+        self.inner_execute(check)
+            .unwrap_or_else(CheckResult::Failed)
+    }
+    fn get_json(&self) -> serde_json::Value {
+        serde_json::to_value(self).unwrap()
+    }
+}
+
+impl ContainerLocalTime {
+    fn inner_execute(&mut self, check: &mut Check) -> Result<CheckResult, failure::Error> {
+        let docker_host_arg = if let Some(docker_host_arg) = &check.docker_host_arg {
+            docker_host_arg
+        } else {
+            return Ok(CheckResult::Skipped);
+        };
+
+        let expected_duration = std::time::SystemTime::now()
+            .duration_since(std::time::UNIX_EPOCH)
+            .context("Could not query local time of host")?;
+        self.expected_duration = Some(expected_duration.to_owned());
```

Comment:
`self.expected_duration = Some(expected_duration);`

`std::time::Duration` is `Copy`.